### ISSUE TITLE
WWW-2019-Learning Dual Retrieval Module for Semi-supervised Relation Extraction

### ISSUE THREAD

**Issue by BrambleXu** [BrambleXu/knowledge-graph-learning#187]
**一句话总结：**

针对semi-supervised relation extraction问题，提出了将RE模型和retrieval module 进行联合训练的方法。

> 问题：基于weak supervision (标注数据少) 的RE模型 (self-training, multi-view) 有两个问题, suffer from the issue of semantic drift, not fully capture the problem characteristics of RE
> 提案：we leverage a key insight that retrieving sentences expressing a relation is a dual task of predicting relation label for a given sentence— two tasks are complementary to each other and can be optimized jointly for mutual enhancement.
> 具体做法：讲一个retrieval module 和原来的RE module进行联合训练。
> 效果：retrieval module能从unlabeled data中找到high-quality样本，来提高RE效果，反之，RE学到的知识也能提升retrieval module的效果

**资源：**

- [pdf](https://arxiv.org/abs/1902.07814), [ppt](https://docs.google.com/presentation/d/15TLIQOWhMa3MCkDrCiBUz2r2HBJx9P79R-EmouXsJB4/edit#slide=id.p)
- [code](https://github.com/ink-usc/DualRE)
- [paper-with-code](

**论文信息：**

- Author: University of Southern California
- Dataset: 
- keywords: relation extraction, dual learning, semi-supervised learning

**笔记：**

![image](https://user-images.githubusercontent.com/10768193/57975922-41e2ee00-7a0f-11e9-9cd3-cf611c7e68c0.png)
(什么是RE)

近年来讲DS用于RE的情况增多。但有两个缺点：
- not all sentences where two entities co-occur are expressing their KB relations (i.e., contextagnostic label noise).
- in some application domains the target relations may not be found in existing knowledge bases (e.g., WikiData), making it hard to apply distant supervision for label generation

因为这些问题，导致了如何减少对标注数据依赖的研究，即semi-supervised relation extraction方法，即利用一小部分标注数据以及一大堆未标注数据来训练。这种方法有很广的用例，比如cold-start knowledge base construction2 and question answering

本文研究的问题就是 semi-supervised relation extraction。

针对这个问题，当前有两种流行方法：self-ensembling methods, and self-training methods. 

- self-ensembling： impose the assumption of “consistency under perturbation”—the prediction on the unlabeled instances should remain unchanged under small perturbations either on the data, or on the model parameters. （简单说就是在小规模标注数据上学到的模型，在未标注数据上应该有一致的表现。嗯。。。最近那篇LDS的文章对这个假设可以说很致命啊）

- self-training methods ：aim to incrementally promote high-confidence predictions over unlabeled instances into labeled training set, and improve the model by re-training with the updated labeled set.（在未标注数据上提高高置信度样本的比例，然后添加到标注样本里，再对模型训练）

但是这两个方法有缺点：
1.  Insufficient Supervision: The process of enforcing prediction consistency between model variants in self-ensembling methods does not involve promoting new labeled instances from the unlabeled set. This makes it hard to improve
models when “consistency-based” supervision is relatively weak compared to that from actual instances （说白了就是有标注的样本太少，学到的模型无法模拟真实的大量样本）
2. Semantic Drift [9]: While the self-training methods annotates unlabeled sentences as extra training data, the incremental promotion process can be biased (i.e., provides incorrect predictions), and may accumulate errors (i.e., including falsely-labeled instances into training set). （其实就是 error propagation）

直觉上讲，retrieving sentences for a given relation的dual problem是 predicting the relation expressed in a sentence。具体见Fig 2右边部分

![image](https://user-images.githubusercontent.com/10768193/57976010-47d9ce80-7a11-11e9-96f6-0a33c09a8b0d.png)

**模型图：**

**结果**：

**接下来要看的论文**：
